src/geome/iterables/to_category_iterator.py

```diff
@@ -21,12 +21,12 @@ class ToCategoryIterator(ToIterable):
     axis (int | str): The axis along which to iterate over the categories. Can be either 0, 1, "obs" or "var".
         0 or "obs" means the categories are in the observation axis.
         1 or "var" means the categories are in the variable axis.
-    preserve_categories (bool): Preserves the categories in the resulting AnnData obs and var Series if `preserve_categories` is True.
+    preserve_categories (list): If not None, preserves the indicated categories from the Anndata 'obs' and 'var'
     """
 
     category: str
     axis: Literal[0, 1, "obs", "var"] = "obs"
-    preserve_categories: bool = True
+    preserve_categories: Optional[list[str]] = None
```

Comment:
I like this idea but I think it might be better if we do it like this
```
preserve_categories: Optional[list[str]] = []
```
So empty list by default which would mean no preservation. And if it is `None` that would mean all categories. So then it would make sense to change the variable name to `preserved_categories`. Also to make the typing more modern finally it would look like this
```python
preserved_categories: Sequence[str] | None = tuple()
```
here default is tuple instead because its not good to have default arguments as mutuable objects. Could you also update docstring with that?